--- original/src/main/java/org/apache/commons/math3/ode/nonstiff/RungeKuttaIntegrator.java
+++ refactored/src/main/java/org/apache/commons/math3/ode/nonstiff/RungeKuttaIntegrator.java
@@ -146,14 +146,7 @@
 
       }
 
-      // estimate the state at the end of the step
-      for (int j = 0; j < y0.length; ++j) {
-          double sum    = b[0] * yDotK[0][j];
-          for (int l = 1; l < stages; ++l) {
-              sum    += b[l] * yDotK[l][j];
-          }
-          yTmp[j] = y[j] + stepSize * sum;
-      }
+      estimateState(y, y0.length, stages, yDotK, yTmp);
 
       // discrete events handling
       interpolator.storeTime(stepStart + stepSize);
@@ -185,4 +178,14 @@
 
   }
 
-}
+  private void estimateState(final double[] y, final int dimension, final int stages, final double[][] yDotK, final double[] yTmp) {
+      for (int j = 0; j < dimension; ++j) {
+          double sum = b[0] * yDotK[0][j];
+          for (int l = 1; l < stages; ++l) {
+              sum += b[l] * yDotK[l][j];
+          }
+          yTmp[j] = y[j] + stepSize * sum;
+      }
+  }
+
+}